sys/sys.cpp

```diff
@@ -18,23 +19,58 @@ void operator delete(void *, unsigned int)
     for (;;);
 }
 
-
+// TODO: move this to toolchain-dependent module
 #if UINT32_MAX == UINTPTR_MAX
 #define STACK_CHK_GUARD 0xe2dee396
 #else
 #define STACK_CHK_GUARD 0x595e9fbd94fda766
 #endif
 
+// TODO: move this to toolchain-dependent module
 __attribute__((used))
 uintptr_t __stack_chk_guard = STACK_CHK_GUARD;
 
+// TODO: move this to toolchain-dependent module
 extern "C" __attribute__((noreturn)) __attribute__((used))
 void __stack_chk_fail(void)
 {
     ecl::cout << "Fail!!!" << ecl::endl;
     for(;;);
 }
 
+// TODO: move this to toolchain-dependent module
+extern "C" __attribute__((used))
+int atexit (void (*func)(void))
+{
+    (void) func;
+    return 0;
+}
+
+// TODO: move this to toolchain-dependent module
+extern "C" __attribute__((used))
+int __cxa_guard_acquire(int *gv)
+{
+    // Disable interrupts to prevent concurent access
+    ecl::disable_irq();
+
+    if ((*gv) & 1) {
+        // Already locked
+        return 0;
+    } else {
+        *gv |= 1;
```

Comment:
Wouldn't using of 1 and 0 be faster?
```c
if (*gv) {
    return 0;
} else {
    *gv = 1;
    return 1;
}
```

I haven't dug in assembly but that seems faster.